var/spack/repos/builtin/packages/dftbplus/package.py

```diff
@@ -160,3 +233,56 @@ def edit(self, spec, prefix):
             )
 
             mconfig.filter("WITH_DFTD3 := .*", "WITH_DFTD3 := 1")
+
+    def cmake_args(self):
+        args = [
+            self.define_from_variant("WITH_OPENMP", "openmp"),
+            self.define_from_variant("WITH_API", "api"),
+            self.define_from_variant("WITH_ARPACK", "arpack"),
+            self.define_from_variant("WITH_CHIMES", "chimes"),
+            self.define_from_variant("WITH_ELSI", "elsi"),
+            self.define_from_variant("WITH_GPU", "gpu"),
+            self.define_from_variant("WITH_MBD", "mbd"),
+            self.define_from_variant("WITH_MPI", "mpi"),
+            self.define_from_variant("WITH_PLUMED", "plumed"),
+            self.define_from_variant("WITH_POISSON", "poisson"),
+            self.define_from_variant("WITH_PYTHON", "python"),
+            self.define_from_variant("WITH_SDFTD3", "sdftd3"),
+            self.define_from_variant("WITH_SOCKETS", "sockets"),
+            self.define_from_variant("WITH_TBLITE", "tblite"),
+            self.define_from_variant("WITH_TRANSPORT", "transport"),
+            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+        ]
+        # SCALAPACK
+        if "+mpi" in self.spec:
+            args.extend(
+                [
+                    self.define("SCALAPACK_FOUND", "true"),
+                    self.define("SCALAPACK_INCLUDE_DIRS", self.spec["scalapack"].prefix.include),
+                    self.define("SCALAPACK_LIBRARIES", self.spec["scalapack"].libs.joined(";")),
+                    self.define("SCALAPACK_LIBRARY", self.spec["scalapack"].libs.joined(";")),
+                ]
+            )
```

Comment:
Please correct me if im wrong but we probably need to define the blas and lapack prefixes for the serial variant 
```suggestion
        if "+mpi" in self.spec:
            # we use scalapack for linear algebra
            args.extend(
                [
                    self.define("SCALAPACK_FOUND", "true"),
                    self.define("SCALAPACK_INCLUDE_DIRS", self.spec["scalapack"].prefix.include),
                    self.define("SCALAPACK_LIBRARIES", self.spec["scalapack"].libs.joined(";")),
                    self.define("SCALAPACK_LIBRARY", self.spec["scalapack"].libs.joined(";")),
                ]
            )
        else:
            # we define the lapack and blas libraries
            lapack_libs = self.spec["lapack"].libs.joined(";")
            blas_libs = self.spec["blas"].libs.joined(";")
            args.extend(
                [
                    self.define("LAPACK_FOUND", True),
                    self.define("LAPACK_INCLUDE_DIRS", spec["lapack"].prefix.include),
                    self.define("LAPACK_LIBRARIES", lapack_libs),
                    self.define("BLAS_FOUND", True),
                    self.define("BLAS_INCLUDE_DIRS", spec["blas"].prefix.include),
                    self.define("BLAS_LIBRARIES", blas_libs),
                    self.define("BLAS_LIBRARY", blas_libs),
                ]
            )
```